### PR TITLE
Generalize `graphnet.deployment` to be compatible with other experiments

### PR DESCRIPTION
**Pull request by RasmusOrsoe**
Addresses #472 by refactoring our existing deployment modules. 

This new structure makes it easier for new experiments to integrate their deployment needs into GraphNeT.

Primarily, the changes are:

- Move generic, experiment-agnostic functionality in `GraphNeTI3Deployer` into a base class called `Deployer`.
- Move generic, experiment-agnostic functionality from `GraphNeTI3Module` into `DeploymentModule`. 

These two classes now live in `graphnet.deployment`. I have added a deprecation warning for `GraphNeTI3Deployer` 

Our existing IceCube-specific implementations now live in `graphnet.deployment.icecube`.

### CODE REVIEW

File: src/graphnet/deployment/icecube/inference_module.py [hunk 1]
```diff
@@ -0,0 +1,207 @@
...
+                " - padding {self.prediction_columns} with NaN."
+            )
+            predictions = np.repeat(
+                [np.nan], len(self.prediction_columns)
+            ).reshape(-1, len(self.prediction_columns))
+        return predictions
+
+    def _create_data_representation(
+        self, frame: I3Frame
+    ) -> Data:  # py-l-i-n-t-:- -d-i-s-able=invalid-name
```
Comment (Aske-Rosted) [inline src/graphnet/deployment/icecube/inference_module.py:154]: Is this some auto-correction gone wrong or is the tag supposed to be like that?

**Review by Aske-Rosted — COMMENTED** [review]
Some small questions/corrections, otherwise looks fine.

File: src/graphnet/deployment/icecube/cleaning_module.py [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+            gcd_file: path to associated gcd file.
+            threshold: the threshold for being considered a positive case.
+                        E.g., predictions >= threshold will be considered
+                        to be signal, all else noise.
+            discard_empty_events: When true, this flag will eliminate events
+                            whose cleaned pulse series are empty. Can be used
+                            to speed up processing especially for noise
+                            simulation, since it will not do any writing or
+                            further calculations.
+            prediction_columns: column names for the predictions of the model.
```
[the remark this answers is not in the document]
Comment (RasmusOrsoe) [reply src/graphnet/deployment/icecube/cleaning_module.py]: Hey - thanks for pointing that out. After this new refactor, the argument is indeed useless. It's been removed.

File: src/graphnet/deployment/icecube/inference_module.py (hunk 1 above)
Comment (RasmusOrsoe) [reply src/graphnet/deployment/icecube/inference_module.py]: I believe so. Removed.

**Comment by RasmusOrsoe** [comment]
@Aske-Rosted thanks for your comments! I have now gone through them and addressed them.

**Review by Aske-Rosted — APPROVED** [review]
nice.

**Pull request merged**